From 6cd90394f9be9759c6483e8729f02fc41f9a7ed8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 30 Apr 2012 13:53:58 +0000
Subject: [PATCH 2/5] i18n: Add more TRANSLATORS comments
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

I took a pass through git.pot and added comments to various messages
that could use one, comments still need to be added to a lot of other
things, but you have to start somewhere.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 advice.c    |    3 +++
 bundle.c    |    5 +++++
 commit.c    |    4 ++++
 connected.c |    3 +++
 diff.c      |    1 +
 remote.c    |   10 ++++++++++
 sequencer.c |   10 ++++++++++
 7 files changed, 36 insertions(+)

diff --git a/advice.c b/advice.c
index a492eea..5944abb 100644
--- a/advice.c
+++ b/advice.c
@@ -37,6 +37,9 @@ void advise(const char *advice, ...)
 
 	for (cp = buf.buf; *cp; cp = np) {
 		np = strchrnul(cp, '\n');
+		/* TRANSLATORS: This will be used to prefix each line of an
+		 * advice message, with the format string being the line
+		 */
 		fprintf(stderr,	_("hint: %.*s\n"), (int)(np - cp), cp);
 		if (*np)
 			np++;
diff --git a/bundle.c b/bundle.c
index 8d31b98..f24999e 100644
--- a/bundle.c
+++ b/bundle.c
@@ -86,6 +86,7 @@ int read_bundle_header(const char *path, struct bundle_header *header)
 	int fd = open(path, O_RDONLY);
 
 	if (fd < 0)
+		/* TRANSLATORS: The argument is a path that can't be opened */
 		return error(_("could not open '%s'"), path);
 	return parse_bundle_header(fd, header, path);
 }
@@ -137,6 +138,9 @@ int verify_bundle(struct bundle_header *header, int verbose)
 	struct object_array refs;
 	struct commit *commit;
 	int i, ret = 0, req_nr;
+	/* TRANSLATORS: This is a heading that'll precede a list of commit
+	 * hashes and reference names
+	 */
 	const char *message = _("Repository lacks these prerequisite commits:");
 
 	init_revisions(&revs, NULL);
@@ -412,6 +416,7 @@ int create_bundle(struct bundle_header *header, const char *path,
 		return error(_("pack-objects died"));
 	if (!bundle_to_stdout) {
 		if (commit_lock_file(&lock))
+			/* TRANSLATORS: The argument is a path we can't create */
 			die_errno(_("cannot create '%s'"), path);
 	}
 	return 0;
diff --git a/commit.c b/commit.c
index 9ed36c7..a4ee4b7 100644
--- a/commit.c
+++ b/commit.c
@@ -45,8 +45,12 @@ struct commit *lookup_commit_or_die(const unsigned char *sha1, const char *ref_n
 {
 	struct commit *c = lookup_commit_reference(sha1);
 	if (!c)
+		/* TRANSLATORS: The argument is a reference name */
 		die(_("could not parse %s"), ref_name);
 	if (hashcmp(sha1, c->object.sha1)) {
+		/* TRANSLATORS: The first argument is a refname, the second is
+		 * a commit hash
+		 */
 		warning(_("%s %s is not a commit!"),
 			ref_name, sha1_to_hex(sha1));
 	}
diff --git a/connected.c b/connected.c
index 1e89c1c..cdd9813 100644
--- a/connected.c
+++ b/connected.c
@@ -45,6 +45,9 @@ int check_everything_connected(sha1_iterate_fn fn, int quiet, void *cb_data)
 		memcpy(commit, sha1_to_hex(sha1), 40);
 		if (write_in_full(rev_list.in, commit, 41) < 0) {
 			if (errno != EPIPE && errno != EINVAL)
+				/* TRANSLATORS: The argument is a strerror()
+				 * message
+				 */
 				error(_("failed write to rev-list: %s"),
 				      strerror(errno));
 			err = -1;
diff --git a/diff.c b/diff.c
index 5d06375..0d4534a 100644
--- a/diff.c
+++ b/diff.c
@@ -3436,6 +3436,7 @@ static int parse_dirstat_opt(struct diff_options *options, const char *params)
 {
 	struct strbuf errmsg = STRBUF_INIT;
 	if (parse_dirstat_params(options, params, &errmsg))
+		/* TRANSLATORS: The argument is an error message */
 		die(_("Failed to parse --dirstat/-X option parameter:\n%s"),
 		    errmsg.buf);
 	strbuf_release(&errmsg);
diff --git a/remote.c b/remote.c
index b296d17..14e7332 100644
--- a/remote.c
+++ b/remote.c
@@ -1604,12 +1604,18 @@ int format_tracking_info(struct branch *branch, struct strbuf *sb)
 	base = shorten_unambiguous_ref(base, 0);
 	if (!num_theirs)
 		strbuf_addf(sb,
+			/* TRANSLATORS: the first argument is a branch name, the
+			 * second is a count of how many commits we're ahead of it
+			 */
 			Q_("Your branch is ahead of '%s' by %d commit.\n",
 			   "Your branch is ahead of '%s' by %d commits.\n",
 			   num_ours),
 			base, num_ours);
 	else if (!num_ours)
 		strbuf_addf(sb,
+			/* TRANSLATORS: the first argument is a branch name, the
+			 * second is a count of how many commits we're behind it
+			 */
 			Q_("Your branch is behind '%s' by %d commit, "
 			       "and can be fast-forwarded.\n",
 			   "Your branch is behind '%s' by %d commits, "
@@ -1618,6 +1624,10 @@ int format_tracking_info(struct branch *branch, struct strbuf *sb)
 			base, num_theirs);
 	else
 		strbuf_addf(sb,
+			/* TRANSLATORS: the first argument is a branch name, the
+			 * second and third arguments are counts of how many
+			 * commits they've diverged by
+			 */
 			Q_("Your branch and '%s' have diverged,\n"
 			       "and have %d and %d different commit each, "
 			       "respectively.\n",
diff --git a/sequencer.c b/sequencer.c
index f83cdfd..b284cf8 100644
--- a/sequencer.c
+++ b/sequencer.c
@@ -118,8 +118,14 @@ static void write_cherry_pick_head(struct commit *commit, const char *pseudoref)
 	filename = git_path("%s", pseudoref);
 	fd = open(filename, O_WRONLY | O_CREAT, 0666);
 	if (fd < 0)
+		/* TRANSLATORS: The argument is a filename that can't be
+		 * opened for writing
+		 */
 		die_errno(_("Could not open '%s' for writing"), filename);
 	if (write_in_full(fd, buf.buf, buf.len) != buf.len || close(fd))
+		/* TRANSLATORS: The argument is a filename that can't be
+		 * opened for writing
+		 */
 		die_errno(_("Could not write to '%s'"), filename);
 	strbuf_release(&buf);
 }
@@ -157,6 +163,9 @@ static void write_message(struct strbuf *msgbuf, const char *filename)
 	int msg_fd = hold_lock_file_for_update(&msg_file, filename,
 					       LOCK_DIE_ON_ERROR);
 	if (write_in_full(msg_fd, msgbuf->buf, msgbuf->len) < 0)
+		/* TRANSLATORS: The argument is a filename that can't be
+		 * written to
+		 */
 		die_errno(_("Could not write to %s"), filename);
 	strbuf_release(msgbuf);
 	if (commit_lock_file(&msg_file) < 0)
@@ -365,6 +374,7 @@ static int do_pick_commit(struct commit *commit, struct replay_opts *opts)
 		struct commit_list *p;
 
 		if (!opts->mainline)
+			/* TRANSLATORS: The argument is a commit digest */
 			return error(_("Commit %s is a merge but no -m option was given."),
 				sha1_to_hex(commit->object.sha1));
 
-- 
1.7.10.546.gbaa1a.dirty

